File: 08-06-2024_Index_of_an_Extra_Element.py
class Solution:

    def findExtra(self, n, a, b):
        low = 0
        high = n - 2
        index = n - 1

        while (low <= high):
            mid = (low + high) // 2
            if (a[mid] == b[mid]):
                low = mid + 1
            else:
                index = mid
                high = mid - 1

        return index

File: 08-06-2024_Index_of_an_Extra_Element.java
class Solution {
    public int findExtra(int n, int arr1[], int arr2[]) {
        int left = 0, right = n - 2;
        int index = n - 1;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (arr2[mid] == arr1[mid]) {
                left = mid + 1;
            } else {
                index = mid;
                right = mid - 1;
            }
        }
        return index;
    }
}
